### PR TITLE
greenkeeping(ts): fix and update types for tree-view

### PR DESCRIPTION
**Pull request by jayantbh**
#### Description

Updated a couple of types related to TreeView. That's all.
The changes are backward compatible.

#### Scope

- [x] Patch: Bug Fix
- [ ] Minor: New Feature
- [ ] Major: Breaking Change

### CODE REVIEW

**Comment by jayantbh** [comment]
Hmm, not sure what's wrong with the PR Label here.

**Comment by gergelyke** [comment]
just added the label, thanks @jayantbh !

**Comment by gergelyke** [comment]
can you help us why do we need to change the flow typdes?

**Comment by jayantbh** [comment]
Hi @gergelyke!
Don't we want the TS and Flow types to be _similar_?

Like, if the TS type is a generic, the flow type should be too? At least that's what I was thinking.

**Comment by jayantbh** [comment]
Ah, I see the flow tests failed. Lemme fix.

**Comment by jayantbh** [comment]
Hmm, generics with defaults are apparently not supported in flow. Is that correct?

File: src/tree-view/types.js [hunk 1]
```diff
@@ -27,13 +27,13 @@ export type TreeViewOverridesT = {
 
 export type TreeNodeIdT = number | string;
 
-export type TreeNodeT = {
+// eslint-disable-next-line flowtype/no-weak-types
+export type TreeNodeT<T: any> = {
```
Comment (chasestarr) [inline src/tree-view/types.js:31]: Default generics in flow use <T = any> maybe that helps? Colon only constrains to a subtype specified. I'm not running the change locally, so not sure if this resolves the flow check

**Comment by jayantbh** [comment]
So, I changed how I was declaring the flow type a generic, and refactored all usages of that type.

**Pull request merged**